corehq/apps/locations/fixtures.py

```diff
@@ -67,76 +74,98 @@ def __call__(self, restore_user, version, last_sync=None, app=None):
-    parent = location.parent
-    parent_type = parent.location_type if parent else None
-    return parent_type == location.location_type.parent_type
+def _all_locations(user):
+    if toggles.SYNC_ALL_LOCATIONS.enabled(user.domain):
+        return LocationSet(SQLLocation.active_objects.filter(domain=user.domain))
+    else:
+        all_locations = set()
+
+        user_location = user.sql_location
+        user_locations = set([user_location]) if user_location is not None else set()
+        user_locations |= {location for location in _gather_multiple_locations(user)}
+
+        for user_location in user_locations:
+            location_type = user_location.location_type
+            expand_from = location_type.expand_from or location_type
+            expand_to = location_type.expand_to
+            expand_from_locations = _get_expand_from_level(user.domain, user_location, expand_from)
+
+            for expand_from_location in expand_from_locations:
+                for child in _get_children(user.domain, expand_from_location, expand_to):
+                    # Walk down the tree and get all the children we want to sync
+                    all_locations.add(child)
+
+                for ancestor in expand_from_location.get_ancestors():
+                    # We sync all ancestors of the highest location
+                    all_locations.add(ancestor)
+
+        return LocationSet(all_locations)
 
 
-def _location_footprint(locations):
+def _gather_multiple_locations(user):
+    """If the project has multiple locations enabled, returns all the extra
+    locations the user is assigned to.
     """
-    Given a list of locations, generate the footprint of those by walking up parents.
+    if user.project.supports_multiple_locations_per_user:
+        location_ids = [loc.location_id for loc in user.locations]
+        for location in SQLLocation.active_objects.filter(location_id__in=location_ids):
+            yield location
 
-    Returns a dict of location ids to location objects.
+
+def _get_expand_from_level(domain, user_location, expand_from):
+    """From the users current location, returns the highest location they want to start expanding from
     """
-    all_locs = LocationSet(locations)
-    queue = list(locations)
-    while queue:
-        loc = queue.pop()
-
-        if loc.location_id not in all_locs:
-            # if it's not in there, it wasn't valid
-            continue
-
-        parent = loc.parent
-        if (parent and
-                parent.location_id not in all_locs and
-                _valid_parent_type(loc)):
-            all_locs.add_location(parent)
-            queue.append(parent)
-
-    return all_locs
+    if user_location.location_type.expand_from_root:
+        return SQLLocation.root_locations(domain=domain)
+    else:
+        ancestors = (
+            user_location
+            .get_ancestors(include_self=True)
+            .filter(location_type=expand_from, is_archived=False)
+        )
```

Comment:
@proteusvacuum and I offlined about this, but this should only ever return one location unless something has gone wrong